File: robin/python/programmers/level1/study4/count_sum_divisor.py
def solution(left, right):
    answer = 0
    divisor = 0
    for val in range(left, right + 1):

        for num in range(1, val + 1):

            if val % num == 0:
                divisor += 1

        if divisor % 2 == 0:
            answer += val
        else:
            answer -= val

        divisor = 0

    return answer


def solution1(left, right):
    answer = 0
    for i in range(left, right + 1):
        if int(i ** 0.5) == i ** 0.5:
            answer -= i
        else:
            answer += i
    return answer


def solution3(left, right):
    import math
    answer = 0
    for i in range(left, right + 1, 1):
        sqrt = math.sqrt(i)
        if int(sqrt) == sqrt:
            answer -= i
        else:
            answer += i

    return answer


if __name__ == '__main__':
    left = 13
    right = 17
    print(solution(left, right))
    print(solution1(left, right))
    print(solution3(left, right))

File: robin/java/src/programmers/level1/study4/CountSumDivisor.java
package programmers.level1.study4;

public class CountSumDivisor {
    public static int solution(int left, int right) {
        int answer = 0;

        for (int i = left; i < right + 1; i++) {

            if ((int) Math.sqrt(i) == Math.sqrt(i)) {
                answer -= i;
            } else {
                answer += i;
            }
        }
        return answer;
    }

    public static void main(String[] args) {
        int left = 13;
        int right = 17;

        System.out.println(solution(left, right));
    }
}
